dev_guide/app_tutorials/ruby_on_rails.adoc

```diff
@@ -0,0 +1,357 @@
+
+=== Configuring the Application For OpenShift
+
+In order to have your application communicating with the PostgreSQL database service that will be running in OpenShift, you will need to edit the `default` section in your `config/database.yml` to use link:../../db_images/postgresql.html#environment-variables[environment variables], which you will define later, upon the database service creation.
+
+The `default` section in your edited `config/database.yml` together with pre-defined variables should look like:
+
+----
+<% user = ENV.key?("POSTGRESQL_ADMIN_PASSWORD") ? "root" : ENV["POSTGRESQL_USER"] %>
+<% password = ENV.key?("POSTGRESQL_ADMIN_PASSWORD") ? ENV["POSTGRESQL_ADMIN_PASSWORD"] : ENV["POSTGRESQL_PASSWORD"] %>
+<% db_service = ENV.fetch("DATABASE_SERVICE_NAME","").upcase %>
+
+default: &default
+  adapter: postgresql
+  encoding: unicode
+  # For details on connection pooling, see rails configuration guide
+  # http://guides.rubyonrails.org/configuring.html#database-pooling
+  pool: <%= ENV["POSTGRESQL_MAX_CONNECTIONS"] || 5 %>
+  username: <%= user %>
+  password: <%= password %>
+  host: <%= ENV["#{db_service}_SERVICE_HOST"] %>
+  port: <%= ENV["#{db_service}_SERVICE_PORT"] %>
+  database: <%= ENV["POSTGRESQL_DATABASE"] %>
+----
+
+For an example of how the final file should look, check out link:https://github.com/openshift/rails-ex[Ruby on Rails example application] link:https://github.com/openshift/rails-ex/blob/master/config/database.yml[config/database.yml].
+
+=== Store your App in Git
+
+OpenShift requires link:http://git-scm.com/[git], if you don't have it installed you will need to install it.
+
+Building an application in OpenShift usually requires that the source code be stored in a link:http://git-scm.com/[git] repository, so you will need to install `git` if you do not already have it.
+
+Make sure you are in your Rails application directory by running `ls -1` command. The output of the command should look like:
+
+----
+$ ls -1
+app
+bin
+config
+config.ru
+db
+Gemfile
+Gemfile.lock
+lib
+log
+public
+Rakefile
+README.rdoc
+test
+tmp
+vendor
+----
+
+Now run these commands in your Rails app directory to initialize and commit your code to git:
+
+----
+$ git init
+$ git add .
+$ git commit -m "initial commit"
+----
+
+Once your application is committed you need to push it to a remote repository. For this you would need a link:https://github.com/join[GitHub account], in which you link:https://help.github.com/articles/creating-a-new-repository/[create a new repository].
+
+Set the remote that points to your `git` repository:
+
+----
+$ git remote add origin git@github.com:<namespace/repository-name>.git
+----
+
+After that, push your application to your remote git repository.
+
+----
+$ git push
+----
+
+== Deploy your application to OpenShift
+
+To deploy your Ruby on Rails application we will need to create a new link:../projects.html[Project] in which our application will live, by typing:
```

Comment:
Suggest: "To deploy your Ruby on Rails application, create a new link...[Project] for the application:"